src/deadline/job_attachments/_incremental_downloads/_manifest_s3_downloads.py

```diff
@@ -213,16 +215,35 @@ def _download_manifest_and_make_paths_absolute(
     )
     # Convert all the manifest paths to have absolute normalized local paths
     for manifest_path in manifest.paths:
-        manifest_path.path = os.path.normpath(os.path.join(root_path, manifest_path.path))
-        # TODO: Apply path mapping rules to manifest_path.path right here
+        original_path = manifest_path.path
+        if (
+            manifest_file_system_location_name
+            and manifest_file_system_location_name in local_file_system_locations
+        ):
```

Comment:
It looks like this transformation will only work when the file system location names from storage profiles and the manifest root paths line up precisely. I believe the current implementation works this way, but that isn't a requirement of the system, so we should make the path mapping code more general. The way to generalize it is to produce something to perform path mappings, such as a list of path mapping rules, in the per-job code since each job has a storage profile associated with the submitting workstation. Then the parameter that goes in here is the path mapping object.